### PR TITLE
Update base image to CBL-D

### PR DESCRIPTION
**Pull request by darrentu**
We need to move from Ubuntu 16.04 because it is approaching end of support, and we want newer versions of many tools (such as Python) which are included in the base distribution. 

CBL-D is not a standalone Linux distribution, but tracks Debian stable very closely. The primary difference between Debian and CBL-D is that Microsoft compiles all the packages included in the CBL-D repository internally. This helps guard against supply chain attacks (https://en.wikipedia.org/wiki/Supply_chain_attack). Many other Microsoft services use this image.

Overall we think Debian stable is a good match for Cloud Shell's requirements, and using the CBL-D repository gives us some added security.

This update also refreshes most tools included in the base image, including chef, ansible, batch-shipyard, and more.

### CODE REVIEW

**Review by Cheneric95 — APPROVED** [review]

**Review by farehar — APPROVED** [review]

File: linux/tools.Dockerfile [hunk 1]
```diff
@@ -5,16 +5,16 @@
 
 # To build yourself locally, override this location with a local image tag. See README.md for more detail
 
-ARG IMAGE_LOCATION=cdpxlinux.azurecr.io/artifact/b787066e-c88f-4e20-ae65-e42a858c42ca/official/azure/cloudshell:1.0.20200727.1.base.master.40d4519a
+ARG IMAGE_LOCATION=cdpxlinux.azurecr.io/artifact/b787066e-c88f-4e20-ae65-e42a858c42ca/buddy/azure/cloudshell:1.0.20201009.1.base.add-cbld.5113b5b1
```
Comment (BellaLi) [inline linux/tools.Dockerfile:8]: Do we want to load from this buddy version?

File: linux/tools.Dockerfile [hunk 2]
```diff
@@ -5,16 +5,16 @@
...
 
 # Copy from base build
 FROM ${IMAGE_LOCATION}
 
 # Install latest Azure CLI package. CLI team drops latest (pre-release) package here prior to public release
 # We don't support using this location elsewhere - it may be removed or updated without notice
-RUN wget -nv https://azurecliprod.blob.core.windows.net/cloudshell-release/azure-cli-latest.deb \
-  && dpkg -i azure-cli-latest.deb \
-  && rm -f azure-cli-latest.deb
+RUN wget -nv https://azurecliprod.blob.core.windows.net/cloudshell-release/azure-cli-latest-buster.deb \
```
Comment (BellaLi) [inline linux/tools.Dockerfile:15]: Curious: what's the buster version

**Review by BellaLi — APPROVED** [review]

File: linux/tools.Dockerfile (hunk 1 above)
Comment (edyoung) [reply linux/tools.Dockerfile:8]: Had to merge to master to kick off a new official build. Have now updated this to point to that build. A bit of a messy process...

File: linux/tools.Dockerfile (hunk 2 above)
Comment (edyoung) [reply linux/tools.Dockerfile:15]: Azure-cli has packages for different linux distros. This is the one which works with Debian (closest match to CBL-D). Contains the same version as the other package.

**Pull request merged**